modules/apps/knowledge-base/knowledge-base-service/src/main/java/com/liferay/knowledge/base/service/impl/KBArticleLocalServiceImpl.java

```diff
@@ -1764,6 +1795,56 @@ private List<KBArticle> _getKBArticlesByCompanyIdAndExpirationDate(
 			));
 	}
 
+	private List<KBArticle> _getKBArticlesByCompanyIdAndReviewDate(
+		long companyId, Date reviewDateGT, Date reviewDateLTE) {
+
+		return kbArticlePersistence.dslQuery(
+			DSLQueryFactoryUtil.select(
+				KBArticleTable.INSTANCE
+			).from(
+				KBArticleTable.INSTANCE
+			).where(
+				KBArticleTable.INSTANCE.companyId.eq(
+					companyId
+				).and(
+					KBArticleTable.INSTANCE.reviewDate.gt(reviewDateGT)
+				).and(
+					KBArticleTable.INSTANCE.reviewDate.lte(reviewDateLTE)
+				)
+			));
+	}
+
+	private String _getKBArticleURL(KBArticle kbArticle)
+		throws PortalException {
+
+		String controlPanelFullURL = _portal.getControlPanelFullURL(
```

Comment:
have you tried to use 
```
return PortletURLBuilder.create(
			_portal.getControlPanelPortletURL(
				httpServletRequest, 
				...
```

Reply:
Before my changes, there method `_getKBArticleURL` was using the `PortletURLBuilder` as you suggest. However, this builder needs an `HTTPServletRequest`, which doesn't exist in the new review action.